ignite/engine/__init__.py

```diff
@@ -29,6 +30,10 @@ def create_supervised_trainer(model, optimizer, loss_fn,
             with respect to the host. For other cases, this argument has no effect.
         prepare_batch (callable, optional): function that receives `batch`, `device`, `non_blocking` and outputs
             tuple of tensors `(batch_x, batch_y)`.
+        output_transform (callable, optional): function that receives 'x', 'y', 'y_pred', 'loss' and returns value
+            to be assigned to engine's state.output after each iteration. Default is returning loss.item(). If you
+            are going to attach metrics to trainer, you should pass
```

Comment:
nit: I would move "If you are going to attach metrics to trainer, you should pass `output_transform = lambda x, y, y_pred, loss: (y_pred, y,)`" to `create_supervised_evaluator`. We are not intended to attach metrics to the trainer as metrics as somewhat running averages of running model.

Reply:
No need to move it to create_supervised_evaluator because it's already OK there by default.
Changing output to (y_pred, y,) is required here and main motivation for it is attaching metrics to trainer. If loss calculation is time consuming, calculating metrics on train set while trainig is better choice then calculating them using evaluator because it means repeated run on the train set. Although the second option is more precise of cause, it's cost can be too high.

Reply:
@IlyaOvodov we do not encourage attaching metrics to the trainer : https://pytorch.org/ignite/master/quickstart.html#id1 
So, please remove this phrase. 
And in case of `create_supervised_evaluator` we can just recall that to compute metrics we need to return also `y_pred` and `y`.

Reply:
Done. Also added comment on 'create_supervised_evaluator'.

Reply:
By the way. What if to change default output_transform for metrics from lambda x: x to lambda x: x[:2]?
As metrics now experct tuple (y_pred, y) it will not change current functioning but extend it: tuple of (y_pred, y, somth else,...) will become acceptable, so adding some information to state.output using means of this PR will not require using output_transform in metrics.

Reply:
Good idea, maybe something like `lambda x: x[:2] if len(x) > 2 else x` just to ensure

Reply:
I think in v0.3.0 we should simplify this stuff and write a dict like `{'y_pred': y_pred, 'y': y}` to the `state.output`

Reply:
:) 
Need to think about it. From one point of view it's good (I started with it), but somebody can use some custom metric inherited fromMetric class and expect to receive that output that he returned from process function, proposed sofistication will be suurprise...